static/js/containers/pages/register/RegisterEmailPage.js

```diff
@@ -52,36 +54,35 @@ export class RegisterEmailPage extends React.Component<Props> {
     } = this.props
 
     try {
-      const {
-        body: { state, errors }
-      }: { body: AuthResponse } = await registerEmail(email, recaptcha, next)
+      const { body } = await registerEmail(email, recaptcha, next)
 
-      if (state === STATE_REGISTER_CONFIRM_SENT) {
-        addUserNotification({
-          "email-sent": {
-            type:  ALERT_TYPE_TEXT,
-            props: {
-              text: emailNotificationText(email)
+      handleAuthResponse(history, body, {
+        [STATE_REGISTER_CONFIRM_SENT]: () => {
+          addUserNotification({
+            "email-sent": {
+              type:  ALERT_TYPE_TEXT,
+              props: {
+                text: emailNotificationText(email)
+              }
             }
-          }
-        })
-        history.push(routes.login.begin)
-      } else if (state === STATE_LOGIN_PASSWORD) {
-        addUserNotification({
-          "account-exists": {
-            type:  ALERT_TYPE_TEXT,
-            color: "danger",
-            props: {
-              text: accountExistsNotificationText(email)
+          })
+        },
+        [STATE_LOGIN_PASSWORD]: () => {
+          addUserNotification({
+            "account-exists": {
+              type:  ALERT_TYPE_TEXT,
+              color: "danger",
+              props: {
+                text: accountExistsNotificationText(email)
+              }
             }
-          }
-        })
-        history.push(routes.login.password)
-      } else if (errors.length > 0) {
-        setErrors({
-          email: errors[0]
-        })
-      }
+          })
+          history.push(routes.login.password)
```

Comment:
Looks like this redirect is already being done in `handleAuthResponse` so you can get rid of this

Reply:
Ah, rebasing strikes again, good catch!